src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -208,7 +209,14 @@ internal void BuildAndLogConsoleLog(string msg, LogLevel level)
 
         public void WaitForProcessExitInMilliSeconds(int waitTime)
         {
-            Process.WaitForExit(waitTime);
+            try
+            {
+                Process.WaitForExit(waitTime);
+            }
+            catch (Exception ex)
+            {
+                _workerProcessLogger.LogDebug(ex, "An exception was thrown while waiting for process {processId} to exit. It is possible that the process had already exited and this can be ignored.", Process?.Id);
```

Comment:
catching this exception is the fix; others are just log improvements to help debug in the future